js/using-jquery/drill2.js

```diff
@@ -1,12 +1,36 @@
-$(function(){
+$.fn.hasAttr = function(key) {
+  return this.attr(key) !== undefined;
+};
 
+$(function() {
 
-    console.log('The dom is ready! (drill 2)');
+  console.log('The dom is ready! (drill 2)');
 
-    $(document).on('keyup',function(event){
-        //output to console which key was pressed
-        console.log('The user pressed key: ',event.which);
-    });
+  $(document).on('keyup',function(event){
+    var topMax = 400,
+        leftMax = 800,
+        top = $(this).attr('top');
+        console.log(top);
+        // topNew = topMax - ,
+        leftNew = Math.floor(Math.random() * leftMax);
 
+    //output to console which key was pressed
+    // console.log('The user pressed key: ' + event.which);
+    // switch (event.which) {
+    //   case "37":
+    //     $(".shape").css({ left: leftNew });
+    //     break
+    //   case "38":
+    //     $(".shape").css({ top: topNew });
+    //     break
+    //   case "39":
+    //     $(".shape").css({ left: leftNew });
+    //     break
+    //   case "40":
+    //     $(".shape").css({ top: topNew });
+    //     break
+    //   default
+    // };
+  });
 });
 
```

Comment:
Here is a solution for Drill2
`$(function(){
    console.log('The dom is ready! (drill 2)');
    $('.stage').css({position: 'relative'});
    $('.shape').css({position: 'absolute'});

    $(document).on('keyup',function(event){
        //output to console which key was pressed
        console.log('The user pressed key: ',event.which);
        if (event.which == 37 && $('.shape').position().left > 0) {
          $('.shape').css({left: $('.shape').position().left - 25})
        }
        if (event.which == 39 && $('.shape').position().left < 800) {
          $('.shape').css({left: $('.shape').position().left + 25})
        }
        if (event.which == 38 && $('.shape').position().top > 0) {
          $('.shape').css({top: $('.shape').position().top - 25})
        }
        if (event.which == 40 && $('.shape').position().top < 400) {
          $('.shape').css({top: $('.shape').position().top + 25})
        }
        if (event.which == 83) {
          // $('.shape').animate({left: $('.shape').position.left - 10})
        }
    });

});`